File: views/decor.go
package views

import (
	"html/template"
	"net/http"

	mobiledetect "github.com/Shaked/gomobiledetect"
	"github.com/meehalkoff/loggi"
	"gitlab.com/mvalkov/katzman/lib"
)

func DecorHandler(w http.ResponseWriter, r *http.Request) {

	tmpl := template.New("decor")

	detect := mobiledetect.NewMobileDetect(r, nil)
	if detect.IsMobile() {
		tmpl.ParseFiles(
			"mobile/templates/decor.html",
			"mobile/templates/header.html",
			"mobile/templates/footer.html",
			"mobile/templates/top-panel.html",
			"mobile/templates/calc.html",
			"mobile/templates/call.html",
			"mobile/templates/about.html",
			"mobile/templates/firstPromoBlock.html",
			"mobile/templates/creditPromoBlock.html",
			"mobile/templates/lastPromoBlock.html",
		)
	} else {
		tmpl.ParseFiles(
			"templates/header.html",
			"templates/footer.html",
			"templates/top-panel/top-panel.html",
			"templates/top-panel/about.html",
			"templates/top-panel/calc.html",
			"templates/top-panel/call.html",
			"templates/top-panel/main-menu.html",
			"templates/control-panel.html",
			"templates/firstPromoBlock.html",
			"templates/decor/index.html",
			"templates/decor/summary.html",
			"templates/decor/palitra.html",
			"templates/decor/kachestva.html",
			"templates/decor/schema.html",
			"templates/howWeWork.html",
			"templates/decor/portfolio.html",
			"templates/shared/video.html",
			"templates/decor/promo.html",
			"templates/lastPromoBlock.html",
			"templates/creditPromoBlock.html",
		)
	}
	data := lib.FirstPromoBlock{
		MaterialName: `Декоративные элементы`,
		ID:           "decorPrice",
		PictureURL:   "/assets/img/promo/decor.jpg",
		UnitName:     "м.п.",
		YoutubeID:    "L8aRCQcdznM",
	}
	if err := tmpl.Execute(w, data); err != nil {
		loggi.Fatal(err)
	}
}

File: assets/js/decor.js
var nextArrow = '<div class="control-next-arrow">▶▶</div>';
var prevArrow = '<div class="control-prev-arrow">◀◀</div>';
var maxSlidePosition = 8;

var control = {
    0: {
        id: "first-promo",
        title: "Декоративные элементы: предложение",
        prev: "",
        next: "Обзор" + nextArrow,
    },
    1: {
        id: "summary",
        title: "Декоративные элементы: обзор",
        prev: prevArrow + "Предложение",
        next: "Палитра цветов" + nextArrow,
    },
    2: {
        id: "palitra",
        title: "Декоративные элементы: палитра цветов",
        prev: prevArrow + "Обзор",
        next: "Качества" + nextArrow,
    },
    3: {
        id: "kachestva",
        title: "Декоративные элементы: качества",
        prev: prevArrow + "Палитра цветов",
        next: "Технология монтажа" + nextArrow,
    },
    4: {
        id: "schema",
        title: "Декоративные элементы: монтаж",
        prev: prevArrow + "Качества",
        next: "Портфолио" + nextArrow,
    },
    5: {
        id: "portfolio",
        title: "Декоративные элементы: портфолио",
        prev: prevArrow + "Монтаж",
        next: "Видео обзор" + nextArrow,
    },
    6: {
        id: "video",
        title: "Декоративные элементы: видео обзор",
        prev: prevArrow + "Портфолио",
        next: "Предложение" + nextArrow,
    },
    7: {
        id: "credit-promo",
        title: "Декоративные элементы: рассрочка",
        prev: prevArrow + "Видео",
        next: "Предложение" + nextArrow,
    },
    8: {
        id: "last-promo",
        title: "Декоративные элементы: звоните!",
        prev: prevArrow + "Рассрочка",
        next: "",
    },
}

var maxPortfolioPosition = 8;

var portfolio = {
    0: 'url(/assets/img/decor/portfolio/0.jpg)',
    1: 'url(/assets/img/decor/portfolio/1.jpg)',
    2: 'url(/assets/img/decor/portfolio/2.jpeg)',
    3: 'url(/assets/img/decor/portfolio/3.jpg)',
    4: 'url(/assets/img/decor/portfolio/4.png)',
    5: 'url(/assets/img/decor/portfolio/5.jpg)',
    6: 'url(/assets/img/decor/portfolio/6.jpg)',
    7: 'url(/assets/img/decor/portfolio/7.jpg)',
    8: 'url(/assets/img/decor/portfolio/8.jpg)',
}